apps/src/code-studio/pd/teacher_application/teacher_application.jsx

```diff
@@ -0,0 +1,374 @@
+/* global window */
+
+/**
+ * Teacher Application questionaire
+ */
+
+import React from 'react';
+import _ from 'lodash';
+import {Radio, Checkbox, FormControl, FormGroup, ControlLabel} from 'react-bootstrap';
+
+function FieldGroup({ id, label, ...props }) {
+  return (
+    <FormGroup controlId={id}>
+      <ControlLabel>{label}</ControlLabel>
+      <FormControl {...props} />
+    </FormGroup>
+  );
+}
+
+const grades = ['Kindergarten'].concat(_.range(1,13));
+const subjects = ['Computer Science', 'Computer Literacy', 'Math', 'Science', 'History', 'English', 'Music', 'Art', 'Multimedia', 'Foreign Language'];
+const yesNoResposes = ['Yes', 'No'];
+
+const TeacherApplication = React.createClass({
+
+  getInitialState() {
+    return {
+      selectedCourse: ''
+    };
+  },
+
+  buttonList(type, label, groupName, answers, allowOther = false) {
+    const options = answers.map( (answer, i) => {
+      if (type === 'radio') {
+        return (
+          <Radio
+            value={answer}
+            label={answer}
+            key={i}
```

Comment:
Not necessary, but presuming the values (answers) are unique you could also use the answer as key. It doesn't really matter since these are never reordered.